apps/sharedtest/test/unit/moz_intl_test.js

```diff
@@ -0,0 +1,332 @@
+'use strict';
+
+/* global mozIntl, MockL10n */
+
+require('/shared/test/unit/mocks/mock_l10n.js');
+require('/shared/js/moz_intl.js');
+
+suite('MozIntl', function() {
+  var realL10n;
+
+  suiteSetup(function() {
+    realL10n = navigator.mozL10n;
+    navigator.mozL10n = MockL10n;
+  });
+
+  suiteTeardown(function() {
+    navigator.mozL10n = realL10n;
+  });
+
+  suite('listFormatter', function() {
+    setup(function() {
+      sinon.stub(navigator.mozL10n, 'formatValue', function(...args) {
```

Comment:
I think you might be able to use `this.sinon` here which is a Sinon sandbox created automatically for each test suite: https://github.com/mozilla-b2g/gaia/blob/2bdae877bc22c2abbb080c29f390a4d4b961d677/dev_apps/test-agent/common/test/sinon_helper.js#L4.  You wouldn't need to call `restore` in `teardown`.